### PR TITLE
Feature/signin refactor

### PR DESCRIPTION
**Pull request by vfaltings**
Refactored the sign-in to use Hilt.

To do so, the SwiftAuthenticator interface was created, which represents an authenticator for our app.

### CODE REVIEW

File: app/src/main/java/com/sdp/swiftwallet/common/FirebaseAuthModule.java [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
 @Module
-@InstallIn(ActivityComponent.class)
+@InstallIn(SingletonComponent.class)
 public class FirebaseAuthModule {
+
+    /**
+     * For hilt
+     */
+    @Inject
+    public FirebaseAuthModule(){ }
```
Comment (Zotchh) [inline app/src/main/java/com/sdp/swiftwallet/common/FirebaseAuthModule.java:20]: This has no use here, I tested it on my side but don't touch it for now I did some stuff on another branch to transfer firebaseUtil directly into the hilt module, it makes more sense. So don't bother to change it, I already did so I can just make a small PR after.

File: app/src/main/java/com/sdp/swiftwallet/di/AuthenticatorModule.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import dagger.hilt.components.SingletonComponent;
+
+/**
+ * Hilt Module to inject SwiftAuthenticator
+ */
+@Module
+@InstallIn(SingletonComponent.class)
+public class AuthenticatorModule {
+
+    @Provides
```
Comment (Zotchh) [inline app/src/main/java/com/sdp/swiftwallet/di/AuthenticatorModule.java:18]: is it normal to not have @Singleton tag here ?

**Review by Zotchh — APPROVED** [review]
The login refactoring seems fine, I'm just wondering if there might be better options for some stuff but it's hard to know what is best, we should talk about it next sprint. Otherwise some small stuff but not that important :+1:

File: app/src/androidTest/java/com/sdp/swiftwallet/presentation/signIn/RegisterActivityTest.java [hunk 3]
```diff
@@ -27,29 +27,39 @@
...
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.RuleChain;
 
+@HiltAndroidTest
 public class RegisterActivityTest {
 
-    @Rule
+    @Inject FirebaseAuth db;
```
[the remark this answers is not in the document]
Comment (vfaltings) [reply app/src/androidTest/java/com/sdp/swiftwallet/presentation/signIn/RegisterActivityTest.java:43]: This is from Anders' branch but yeah

File: app/src/androidTest/java/com/sdp/swiftwallet/uiTest/HomeFragmentTest.java [hunk 4]
```diff
@@ -38,6 +41,10 @@
     public Context context;
     public final static int mockCounter = 10;
 
+    @Inject FirebaseAuth firebaseAuth;
```
[the remark this answers is not in the document]
Comment (vfaltings) [reply app/src/androidTest/java/com/sdp/swiftwallet/uiTest/HomeFragmentTest.java:44]: Also not my code

File: app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+    public ActivityScenarioRule<TransactionActivity> testRule = new ActivityScenarioRule<>(TransactionActivity.class);
+    public HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+    @Rule
+    public final RuleChain rule =
+            RuleChain.outerRule(hiltRule).around(testRule);
+
+    @Before
+    public void setupIntent() {
```
[the remark this answers is not in the document]
Comment (vfaltings) [reply app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java:83]: Yeah the name is from the way it was written before, didn't look at the function names when renaming so I didn't catch this

File: app/src/main/java/com/sdp/swiftwallet/common/FirebaseAuthModule.java (hunk 1 above)
Comment (vfaltings) [reply app/src/main/java/com/sdp/swiftwallet/common/FirebaseAuthModule.java:20]: Not my code once again

File: app/src/main/java/com/sdp/swiftwallet/di/AuthenticatorModule.java (hunk 2 above)
Comment (vfaltings) [reply app/src/main/java/com/sdp/swiftwallet/di/AuthenticatorModule.java:18]: Yes, for modules you use @InstallIn(SingletonComponent.class)

File: app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+import dagger.hilt.android.testing.HiltAndroidTest;
+import dagger.hilt.android.testing.UninstallModules;
+import dagger.hilt.components.SingletonComponent;
+
+@UninstallModules(TransactionHistoryProducerModule.class)
+@HiltAndroidTest
+@RunWith(AndroidJUnit4.class)
+public class TransactionActivityTest {
+
+    @Module
```
Comment (ahominal) [inline app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java:47]: Oh you did have time to fix it! nice work

**Review by ahominal — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java (hunk 6 above)
Comment (vfaltings) [reply app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java:47]: Yep, I'm just not sure why the tests aren't passing on the CI still

**Pull request merged**